lib/java/com/google/android/material/bottomsheet/BottomSheetDialog.java

```diff
@@ -467,12 +468,15 @@ private EdgeToEdgeCallback(
       if (backgroundTint != null) {
         // First check for a tint
         lightBottomSheet = isColorLight(backgroundTint.getDefaultColor());
-      } else if (bottomSheet.getBackground() instanceof ColorDrawable) {
-        // Then check for the background color
-        lightBottomSheet = isColorLight(((ColorDrawable) bottomSheet.getBackground()).getColor());
       } else {
-        // Otherwise don't change the status bar color
-        lightBottomSheet = null;
+        Integer backgroundColor = ViewUtils.getBackgroundColor(bottomSheet);
```

Comment:
This doesn't help to add ColorStateListDrawable support, right?

Reply:
Right, it's just a little accompanying refactoring.

Actually, we can do something like this:
```java
ColorStateList backgroundCSL = DrawableUtils.getColorStateListOrNull(bottomSheet.getBackground());
if (backgroundCSL != null) {
  // Then check for the background color
  lightBottomSheet = isColorLight(backgroundCSL.getDefaultColor());
} else {
  // Otherwise don't change the status bar color
  lightBottomSheet = null;
}
```
but I'm not sure how acceptable this solution is to you.

Reply:
Or update `ViewUtils.getBackgroundColor` to handle the CSLDrawable?

Reply:
Ok, if it's acceptable to you, no problem.

Reply:
done